keysign/signature_notifier.go

```diff
@@ -91,8 +90,11 @@ func (s *SignatureNotifier) handleStream(stream network.Stream) {
 	n, ok := s.notifiers[msg.ID]
 	if !ok {
 		logger.Debug().Msgf("notifier for message id(%s) not exist", msg.ID)
+		_ = stream.Close()
```

Comment:
if stream was not closed here, then will it be eventually closed?

Reply:
Yes only once all the goroutines have finished here: 
https://github.com/brewmaster012/go-tss/blob/8f5e91b0ad5b8cd29caa05408edfcf44cd97dfff/tss/keysign.go#L282

The streams from the 3 components get released in the defer line here:
https://github.com/brewmaster012/go-tss/blob/8f5e91b0ad5b8cd29caa05408edfcf44cd97dfff/tss/keysign.go#L214

Reply:
hmm.. any idea as to why we have "deadlock" if the streams get eventually released? 
is it because once stream is exhausted, the "release" functions get deadlocked somehow because no progress can be made otherwhere?

Reply:
I'm not certain that it is exactly a deadlock. It seems like the P2P layer is blocking certain protocols while the streams are open for too long. Even though they are released eventually the lower level blocks the communication protocol.